test/integration/filter_test.rb

```diff
@@ -165,6 +165,36 @@ def test_liquid_argument_error
     assert_match(/\ALiquid error: wrong number of arguments /, exc.message)
     assert_equal(exc.message, Template.parse(source).render)
   end
+
+  def test_sum
+    src_w_prop = "{{ ary | sum: 'qty' }}"
+    src_wo_prop = "{{ ary | sum }}"
```

Comment:
Bunch of nits (non-major):

- I think this would be better with multiple tests

  ```rb
  def test_sum; end
  def test_sum_with_props; end
  def test_sum_non_numeric_values_error; end
  def test_sum_missing_props; end
  ```

  That way, we can inline the source string instead of naming it and it would make the tests more dumb and easier to read. e.g.

  ```rb
  assert_template_result("1000", "{{ numbers | sum }}", { "numbers" => [400, 600] })
  ```


- For consistensy with other tests, the variables holding the strings should be called `source` I think.

- `ary` -> `numbers`, `ary`->`items`
- `qty` -> `quantity`

Reply:
Thanks @charlespwd. I was building the tests in the wrong class. Now that I'm looking in the right place, I see the pattern that you are referencing and I'll revise accordingly.